articles/android-viewpager2/index.md

```diff
@@ -0,0 +1,214 @@
+
+`ViewPager2` is an newer version of the `ViewPager` library. Its major difference to the old library is the use of a `RecyclerView` adapter. With this, views are now recycled. This improves user experience by making smooth transitions. It also minimizes memory usage.
+
+This article goes through implementing `ViewPager2` and `TabLayout` in an android application.
+
+Let's dive in!
+
+### Prerequisites
+To follow through with this tutorial you will need to:
+  1. Have [Android Studio](https://developer.android.com/studio) installed.
+  2. Have a basic knowledge of building Android applications.
+  3. Have a basic understanding of Kotlin programming language.
+
+Let's get started!
+
+### Step 1 — Creating an Android Project
+In this step, we're going to create our application. Open Android Studio and start a new project with an empty activity template. On the next page, give the application a name and keep the default settings.
+
+![app name](/engineering-education/android-viewpager2/app-name.png)
+
+Click `Finish` and wait for the project build process to finish.
+
+### Step 2 — Adding Views in Xml Layouts
+Our `activity_main` layout file contains only two views. Add them as follows.
+
+```xml
+<com.google.android.material.tabs.TabLayout
+    android:id="@+id/tabLayout"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="5dp"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent"/>
+
+<androidx.viewpager2.widget.ViewPager2
+    android:id="@+id/pager"
+    android:layout_width="match_parent"
+    android:layout_height="0dp"
+    android:layout_marginTop="5dp"
+    app:layout_constraintBottom_toBottomOf="parent"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/tabLayout" />
+```
+
+We also need to create a layout that we will display on the `ViewPager2` component. Go to `File -> New -> XML -> Layout XML File` to create the layout file. In the layout file, replace the default `TextView` widget with the one below.
+
+```xml
+<TextView
+    android:id="@+id/textView"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:textColor="@color/black"
+    android:textSize="50sp"
+    tools:text="TextView"
+    app:layout_constraintBottom_toBottomOf="parent"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent" />
+```
+
+We're done with the xml files. Let's go ahead and create an adapter for our `ViewPager2` component.
+
+### Step 3 — Creating ViewPager2 Adapter
```

Comment:
It would be good if you gave a short summary of the function of the adapter to get them up to date.

Reply:
Done